src/main/java/de/tum/cit/aet/artemis/communication/web/SavedPostResource.java

```diff
@@ -0,0 +1,215 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.communication.domain.AnswerPost;
+import de.tum.cit.aet.artemis.communication.domain.Post;
+import de.tum.cit.aet.artemis.communication.domain.Posting;
+import de.tum.cit.aet.artemis.communication.domain.PostingType;
+import de.tum.cit.aet.artemis.communication.domain.SavedPost;
+import de.tum.cit.aet.artemis.communication.domain.SavedPostStatus;
+import de.tum.cit.aet.artemis.communication.dto.PostingDTO;
+import de.tum.cit.aet.artemis.communication.repository.AnswerPostRepository;
+import de.tum.cit.aet.artemis.communication.repository.PostRepository;
+import de.tum.cit.aet.artemis.communication.service.SavedPostService;
+import de.tum.cit.aet.artemis.core.exception.BadRequestAlertException;
+import de.tum.cit.aet.artemis.core.exception.EntityNotFoundException;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+import de.tum.cit.aet.artemis.core.util.TimeLogUtil;
+
+/**
+ * REST controller for managing Message Posts.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class SavedPostResource {
+
+    private static final Logger log = LoggerFactory.getLogger(SavedPostResource.class);
+
+    public static final String ENTITY_NAME = "savedPost";
+
+    private final SavedPostService savedPostService;
+
+    private final PostRepository postRepository;
+
+    private final AnswerPostRepository answerPostRepository;
+
+    public SavedPostResource(SavedPostService savedPostService, PostRepository postRepository, AnswerPostRepository answerPostRepository) {
+        this.savedPostService = savedPostService;
+        this.postRepository = postRepository;
+        this.answerPostRepository = answerPostRepository;
+    }
+
+    /**
+     * GET /saved-posts/{courseId}/{status} : Get saved posts of course with specific status
+     *
+     * @param courseId id of course to filter posts
+     * @param status   saved post status (progress, completed, archived)
+     * @return ResponseEntity with status 200 (Success) if course id and status are ok,
+     *         or with status 400 (Bad Request) if the checks on type or post validity fail
+     */
+    @GetMapping("saved-posts/{courseId}/{status}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<PostingDTO>> getSavedPosts(@PathVariable Long courseId, @PathVariable short status) {
+        log.debug("GET getSavedPosts invoked for course {} and status {}", courseId, status);
+        long start = System.nanoTime();
+
+        SavedPostStatus savedPostStatus;
+        try {
+            savedPostStatus = SavedPostStatus.fromDatabaseKey(status);
+        }
+        catch (IllegalArgumentException e) {
+            throw new BadRequestAlertException("The provided post status could not be found.", ENTITY_NAME, "savedPostStatusDoesNotExist");
+        }
```

Comment:
Wouldn't it make sense to use the actual `SavedPostStatus` enum here? As it's existent on both the client and server, you should be able to use it as a type here, right?

Reply:
I tried, but unfortunately the server magic binding doesn't seem to work here. :( That's why I added this try/catch part.